crowdfunding/users/views.py

```diff
@@ -0,0 +1,114 @@
+from django.http import Http404
+from projects.permissions import IsOwnProfile
+# from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status, generics, permissions
+from rest_framework.permissions import IsAuthenticated
+from django.contrib.auth.models import User
+
+from .models import CustomUser
+from .serializers import CustomUserSerializer, ChangePasswordSerializer, CustomUserDetail
+
+# Create your views here.
+class CustomUserList(generics.ListCreateAPIView):
+    queryset = CustomUser.objects.all()
+    serializer_class = CustomUserSerializer
+    
+    def post(self, request):
+        if request.user.is_authenticated:
+            return Response({"error": "You cannot create a new user while you are logged in."})
+        else:
+            serializer = CustomUserSerializer(data=request.data)
+            if serializer.is_valid():
+                serializer.save()
+                return Response(serializer.data, status=status.HTTP_201_CREATED)
+            else:
+                if 'email' in serializer.errors:
+                    return Response({"error":"This email is associated with another user. Please login or choose an alternative email."}, status=status.HTTP_400_BAD_REQUEST)
+                else:
+                    return Response(
+                        serializer.errors,
+                        status=status.HTTP_400_BAD_REQUEST)
```

Comment:
I'm not sure you needed to add this post method, i see a lot of custom logic here, but also it looks a lot like this is just built in functionality. 
surely this would have been the behavior that would have worked anyway, just by using defaults and writing no code?